rasa/nlu/components.py

```diff
@@ -460,347 +466,75 @@ def name(self) -> Text:
         """
-        pass
+        ...
 
-    def process(self, message: Message, **kwargs: Any) -> None:
-        """Processes an incoming message.
+    @abc.abstractmethod
+    def process(self, messages: List[Message], **kwargs: Any) -> List[Message]:
+        """Processes incoming messages.
 
-        This is the components chance to process an incoming
-        message. The component can rely on
-        any context attribute to be present, that gets created
-        by a call to :meth:`rasa.nlu.components.Component.create`
-        of ANY component and
-        on any context attributes created by a call to
+        This is the components chance to process incoming
+        messages. The component can rely on
+        any context attributes created by a call to
         :meth:`rasa.nlu.components.Component.process`
         of components previous to this one.
 
         Args:
-            message: The :class:`rasa.shared.nlu.training_data.message.Message` to
-                process.
-        """
-        pass
-
-    def persist(self, file_name: Text, model_dir: Text) -> Optional[Dict[Text, Any]]:
-        """Persists this component to disk for future loading.
-
-        Args:
-            file_name: The file name of the model.
-            model_dir: The directory to store the model to.
+            messages: A list of  :class:`rasa.shared.nlu.training_data.message.Message`
+                to process.
 
         Returns:
-            An optional dictionary with any information about the stored model.
+            A list of processed :class:`rasa.shared.nlu.training_data.message.Message`.
         """
-        pass
+        ...
 
-    @classmethod
-    def cache_key(
-        cls, component_meta: Dict[Text, Any], model_metadata: "Metadata"
-    ) -> Optional[Text]:
-        """This key is used to cache components.
-
-        If a component is unique to a model it should return None.
-        Otherwise, an instantiation of the
-        component will be reused for all models where the
-        metadata creates the same key.
-
-        Args:
-            component_meta: The component configuration.
-            model_metadata: The component's :class:`rasa.nlu.model.Metadata`.
-
-        Returns:
-            A unique caching key.
-        """
-        return None
-
-    def __getstate__(self) -> Any:
-        """Gets a copy of picklable parts of the component."""
-        d = self.__dict__.copy()
-        # these properties should not be pickled
-        if "partial_processing_context" in d:
-            del d["partial_processing_context"]
-        if "partial_processing_pipeline" in d:
-            del d["partial_processing_pipeline"]
-        return d
+    @abc.abstractmethod
+    def persist(self) -> None:
+        """Persists this component to disk for future loading."""
+        ...
 
     def __eq__(self, other: Any) -> bool:
         return self.__dict__ == other.__dict__
 
-    def prepare_partial_processing(
```

Comment:
Of this partial processing stuff is now obsolete as we run the pipeline in the graph.